ql/src/java/org/apache/hadoop/hive/ql/queryhistory/repository/IcebergRepository.java

```diff
@@ -101,8 +112,14 @@ protected Table createTable(Hive hive, Database db) throws HiveException {
     return table;
   }
 
+  @VisibleForTesting
+  int getSnapshotMaxAge() {
+    // expire/delete snapshots older than 1 day
+    return 24 * 60 * 60 * 1000;
```

Comment:
that could be set on a table level - `history.expire.max-snapshot-age-ms`

Reply:
oh, yes, that's what you do as well. (not sure if we need HiveConf for that)

Reply:
I would not use a HiveConf, because this only applies when the table is first created, and from that point, the config is not effective (only if I implement some alter table mechanism that overcomplicates this area), so a HiveConf can make the user think the setting make its way magically any time, but in fact, the assumption is that user will ALTER the table manually